gnovm/pkg/gnolang/store.go

```diff
@@ -556,11 +574,47 @@ func (ds *defaultStore) getMemPackage(path string, isRetry bool) *std.MemPackage
 		// Some packages may never be persisted, thus why we only attempt this twice.
 		if !isRetry {
 			if pv := ds.GetPackage(path, false); pv != nil {
+				if false { // XXX
+					fmt.Println("defaultStore.getMemPackage============1 not retry")
+					if ps, ok := ds.iavlStore.(types.Printer); ok {
+						ps.Print()
+					}
+					fmt.Println("defaultStore.getMemPackage============1 not retry ENDPRINT")
+				}
+				// XXX XXX XXX hmmmmmmmm unauthorized error?
+				//ds.iavlStore.(types.WriteThrougher).WriteThrough(1)
+				//ds.baseStore.(types.WriteThrougher).WriteThrough(2)
+				// ds.Flush()
+				// XXX clear cache layers.
+				if cts, ok := ds.iavlStore.(types.ClearThrougher); ok {
+					cts.ClearThrough()
+				}
+				if cts, ok := ds.baseStore.(types.ClearThrougher); ok {
+					cts.ClearThrough()
+				}
```

Comment:
essential change #2, since this transaction's CacheStore wrapper already cached the fact that the mempackage does not exist, we need to clear this non-dirty empty item from the cache in order to be able to fetch the mempackage that was written to the iavltree (see essential change #1).